optimade/server/models/util.py

```diff
@@ -1,10 +1,29 @@
-from typing import cast, Any, Dict, Type
+from typing import cast, Any, Dict, Type, Optional
 
-from pydantic import ConstrainedInt, errors
+from pydantic import ConstrainedInt, errors, BaseModel, Schema, UrlStr
 from pydantic.types import OptionalInt
 from pydantic.validators import list_validator
 
 
+class Maintainer(BaseModel):
+    email: str = Schema(..., description="the maintainer's email address")
```

Comment:
You can use `EmailStr` (from `pydantic`) instead of `str` here.

Furthermore, it seems to me the convention is to concatenate class names, i.e., this should be `ImplementationMaintainer` - however, this seemingly-to-me-apparent convention may most definitely be wrong ;)